External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -291,26 +288,27 @@
 maskNaN(relativeDelta > thresholdIs & distMapInward < edgeDistance) = 1;
 
 %% Extrapolation of values based on the edge values 
-% Calculate the distance map from the non-Nan towards NaN
-[~, distY, distX, distZ] = xASL_im_DistanceTransform(~maskNaN);
+% Calculate the relative distance map from the non-Nan towards NaN. Note that for non-NaNs, these relative coordinates are zero
+[~, distX, distY, distZ] = xASL_im_DistanceTransform(~maskNaN);
 
-% We calculate the meshgrid
+% We calculate the meshgrid - three 3D matrices that give, respectively, for each voxel its X, Y, and Z coordinates.
 [gridX, gridY, gridZ] = ndgrid(1:size(IM, 1), 1:size(IM, 2), 1:size(IM, 3));
 
-% We calculate the coordinates of the border voxel
+% We calculate the coordinates of the border voxel, note that for non-NaNs this gives the original voxel location
```

Comment:
You can slightly improve this for understanding, e.g. it is not "borderX" but "newValuesX" right? (the border is at the border of the non-nans but the non-nans could be halfway the image).

gridX == coordinateX?
distX = distanceX?

Reply:
1. We first calculate the border coordinates (borderX)
2. Then we take the border value and set each voxel to the border value
3. Then we update the value by the distance from border

So borderX is correct, it's not newValuesX
gridX can be indeed renamed to coordinateX
distX can be distanceX % 0 for non-NaNs

Reply:
OK so:

CoordinateBorderX
CoordinateOriginalX
distanceX

? Nice if we can change this to make it more readable (not very important)

Reply:
can do...